dashboard/src/modules/components/ComparisonComponent/PanelContent.jsx

```diff
@@ -1,36 +1,82 @@
 import "./index.less";
 
-import { List, ListItem } from "@patternfly/react-core";
+import { Checkbox, List, ListItem } from "@patternfly/react-core";
+import React, { useCallback, useMemo } from "react";
+import { getQuisbyData, setSelectedId } from "actions/comparisonActions";
 import { useDispatch, useSelector } from "react-redux";
 
-import React from "react";
-import { getQuisbyData } from "actions/comparisonActions";
-
 const PanelConent = () => {
   const dispatch = useDispatch();
   const { datasets } = useSelector((state) => state.overview);
-  const { activeResourceId } = useSelector((state) => state.comparison);
+  const {
+    activeResourceId,
+    isCompareSwitchChecked,
+    selectedResourceIds,
+    searchValue,
+  } = useSelector((state) => state.comparison);
+  const onFilter = useCallback(
+    (item) => {
+      if (searchValue === "") {
+        return true;
+      }
+      let input;
+      try {
+        input = new RegExp(searchValue, "i");
+      } catch (err) {
+        input = new RegExp(
+          searchValue.replace(/[.*+?^${}()|[\]\\]/g, "\\$&"),
```

Comment:
So you're allowing RegEx characters in the user's filter string, but if it doesn't compile you're quoting all special characters and compiling again? I'm not sure this is what I'd want.

* If I use regex characters accidentally, quoting them is good, but probably not as a fallback. That is, if I didn't intentionally construct a regex, then the results of `AB*C` will surprise me;
* If I construct a regex but get it wrong, quoting all special characters will probably still surprise me.

I think that if we're not going to allow regex input, then you should quote (or reject) special characters up front (or simply use a non-regex filter), and if we intentionally allow regex then a compile failure should be an error visible to the user so they can retry.

I also noticed that in the demos you were using this to find the `uperf` benchmark runs. As mentioned earlier, using the name is unreliable here: it'd be better to search for an exact match in `server.benchmark`... although I can imagine users finding name search useful as well, I suspect that in general they'll need a lot more.

I don't think we want to get into this for this PR, but something to keep in mind is that I think perhaps in the longer run what we want is a general search component for datasets that you can use on each page that allows filtering by wildcard (regex?) name match or by arbitrary metadata. That is, when I get to the comparison I probably don't want to see all my/public `uperf` but rather the set of `uperf` data I generated for a specific purpose, which might be based on my config string or a `global.lightspeed.run` value or something.